tests/integration2/galley/conversion/testdata/load.go

```diff
@@ -17,28 +17,53 @@ package testdata
 import (
 	"fmt"
 	"path"
+	"sort"
+	"strconv"
 	"strings"
 )
 
 // Load loads the test data set.
-func Load() ([]TestInfo, error) {
-	var result []TestInfo
+func Load() ([]*TestInfo, error) {
+	infos := make(map[string]*TestInfo)
+
+	// Go through each asset and constructs the test list by considering the input files only.
+	// The other files are checked/verified based on the conventions.
 	for _, asset := range AssetNames() {
 		ext := path.Ext(asset)
-		// Use the input file to drive the generation of TestInfo resources.
-		if ext != ".yaml" {
+
+		// If there is an ignore file, then mark the relevant test as ignored.
```

Comment:
nit: this comment seems out of place.  Also, the comment uses the term "ignore" but the code uses "skip"